internal/broker/broker.go

```diff
@@ -393,86 +393,83 @@ func (b *Broker) IsAuthenticated(sessionID, authenticationData string) (string,
 	if err = b.updateSession(sessionID, session); err != nil {
 		return AuthDenied, "", err
 	}
-	return access, data, nil
+
+	encoded, err := data.MarshalJSON()
```

Comment:
Since the type is unknown here, we can't use `json.Marshal` directly otherwise it would defer to reflection + recursion for each of the structs fields and override our current implementation for them.

Reply:
Now that I reviewed myself, I'll also address my own comment and add it as a comment to the code :joy:

Reply:
Hum, I’m unsure to understand you here.

I was going to suggest `json.Marshal` here too, because it will, for each type, assert if you have the interface and call it, as in https://pkg.go.dev/encoding/json#Marshal: "Marshal traverses the value v recursively. If an encountered value implements [Marshaler](https://pkg.go.dev/encoding/json#Marshaler) and is not a nil pointer, Marshal calls [Marshaler.MarshalJSON] to produce JSON"

So, from what I gues `json.Marshall(data)` should then call `data.MarshalJSON` under the hood?

Reply:
Well, it doesn't seem to apply to the actual full type. Here, when calling `json.Marshal(data)` it would just do the default struct marshaling instead (very weird since both types implement the `json.Marshaler` interface, but that's what happens), so my guess is that it can't assert the type of data by itself.